doc/api/customization.md

```diff
@@ -0,0 +1,51 @@
+# Customization
+
+> Stability: 1.0 - Early development
+
+Node.js has multiple APIs for customizing behaviour and can easily be invoked
+via [`--import`][]: A `"nodejs-customization"` [entry point][`"exports"`] for a
+package loaded via [`--import`][] is automatically run at startup. The
+package.json of such a plugin would look something like this:
+
+```json
+{
+  "name": "example-nodejs-plugin",
+  "keywords": [
+    "nodejs-plugin"
+  ],
+  "exports": {
+    "nodejs-customization": "./registration.mjs"
+  }
+}
+```
+
+Setting the keyword `nodejs-plugin` is important for users to find the package
+(which may be automated). It should also contain the official name(s) for the
+support it provides; for a plugin extending support for typescript would contain
+`typescript` in its keywords; avoid red-herrings such as `typescript` as a
+keyword when the library does not extend support for typescript but is merely
+itself written in typescript. Failing to avoid these red-herrings will cause
+your package to be listed for things it does not do, aggravating users.
+
+These steps configure Node.js with a customization plugin:
```

Comment:
This section mixes instructions for package authors writing a plugin with instructions for using such a plugin. I feel like those should be separate sections, as end users aren’t going to care about how a plugin should be authored.

Reply:
I'll add a heading ("Setting up a plugin") between 28 and 30

Reply:
We should probably put the “Using a plugin” section first, as it applies to more people.